app.go

```diff
@@ -39,6 +41,7 @@ type ReadOnlyState interface {
 	Config() *cctypes.Config
 	EnabledFeatures() []string
 	GetMinBuildNumber() uint64
+	GetTrieDB() *trie.Database
```

Comment:
I don't what the `trie.Database` exposed via `ReadOnlyState`/`State`

Reply:
How about `EVMStateDB`?